pages/analytics/integrations.md.erb

```diff
@@ -0,0 +1,379 @@
+# Integrate your test framework
+
+<div class="Docs__wip-note">
+  <p class="Docs__wip-heading">Test Analytics is an invite-only beta feature. Join the <a href="https://buildkite.com/test-analytics">waitlist</a>.</p>
+</div>
+
+{:toc}
+
+Test Analytics integrates directly with your test runner to provide in-depth information about your tests (including spans) in real time. The beta release ships with [RSpec](#integrating-with-rspec), and you can also upload test results by importing [json](#importing-json) or [JUnit XML](#importing-junit-xml).
+
+## Integrating with RSpec
+
+[RSpec](https://rspec.info/) is a behaviour driven development library for Ruby.
+If you're aleady using RSpec for your tests, add the `rspec-buildkite-analytics` Gem to your code to integrate your test results into your Test Analytics dashboard.
+
+1. Create a new branch:
+
+    ```
+    git checkout -b install-buildkite-test-analytics
+    ```
+
+2. Add `rspec-buildkite-analytics` to your `Gemfile` in the `:test` group:
+
+    ```rb
+    group :test do
+      gem "rspec-buildkite-analytics"
+    end
+    ```
+
+3. Run `bundle` to install the gem and update your `Gemfile.lock`:
+
+    ```sh
+    $ bundle
+    ```
+
+3. Add the Test analytics code to your application in `spec/spec_helper.rb`, and set the envionment variable [securely](/docs/pipelines/secrets) on your agent or agents.
+
+    ```rb
+    require "rspec/buildkite/analytics"
+
+    RSpec::Buildkite::Analytics.configure(ENV["TEST_ANALYTICS_TOKEN"])
+    ```
+
+4. Commit and push your changes:
+
+    ```sh
+    $ git add .
+    $ git commit -m "Install and set up Buildkite Test Analytics"
+    $ git push
+    ```
+
+Once you’re done, in your Test analytics dashboard, you'll see analytics of test executions on all branches that include this code.
+
+### Supported continuous integration platforms
+
+Test Analytics works with
+
+* Buildkite
+* GitHub Actions
+* CircleCI
+
+<!-- Related PR: https://github.com/buildkite/rspec-buildkite-analytics/pull/80 -->
+
+Test Analytics can provide quick links from Test Analytics to your builds, regardless of your CI platform.
+
+<!-- TODO: check with team on how to describe what benefits this provides. I think there are more than just the link to build. -->
+
+### Environment variables
+
+The following envionrment variables are generated by Test Analytics and can be used in your build.
+
+* `BUILDKITE_ANALYTICS_KEY` - a unique key for the build that initiated the Test Analytics run.
+* `BUILDKITE_ANALYTICS_URL` - a URL that links to the build
+* `BUILDKITE_ANALYTICS_BRANCH` - the branch or ref that this run is for
+* `BUILDKITE_ANALYTICS_SHA` - the commit sha for the head of the branch
+* `BUILDKITE_ANALYTICS_NUMBER` - the build number of the build
+* `BUILDKITE_ANALYTICS_JOB_ID` - the id of a job within the build
+* `BUILDKITE_ANALYTICS_MESSAGE` - the commit message for the head of the branch.
+
```

Comment:
These environment variables aren't generated *by* Test Analytics (and there's also a typo in environment!) 

These environment variables need to be provided *to* Test Analytics. In the rspec-buildkite-analytics gem we use the following Buildkite env variables for these values:
https://github.com/buildkite/rspec-buildkite-analytics/blob/77a6f29ef8fe9232e2c13649980a841dd16247ae/lib/rspec/buildkite/analytics/ci.rb#L52-L62

Providing these env variables are important for the full TA experience (like the branch filtering on the pages) but none of them are strictly required by the JUnit or JSON API other than the `BUILDKITE_ANALYTICS_KEY`. I think maybe it needs to be communicated that sensible values need to be used for these if coming from another CI?

Reply:
Cheers @niceking 

I get the suboptimal experience part, it's the same for the JSON ingestor, right? But if you're using RSpec it's sufficient to make sure the envionment variables exist, and they will be picked up, you don't need to pass them directionly, is what we're saying?

Reply:
If you're using RSpec AND Buildkite, yeah they'll get picked up automatically. But if you're using RSpec and another CI tool, then you'll have to pass them/put them in an ENV file/however you deal with ENV variables